pkg/asset/cluster/aws/aws.go

```diff
@@ -87,20 +84,14 @@ func tagSharedVPCResources(ctx context.Context, clusterID string, installConfig
 
 	tagKey, tagValue := sharedTag(clusterID)
 
-	cfg, err := configv2.LoadDefaultConfig(ctx, configv2.WithRegion(installConfig.Config.Platform.AWS.Region))
+	ec2Client, err := awsconfig.NewEC2Client(ctx, awsconfig.EndpointOptions{
+		Region:    installConfig.Config.Platform.AWS.Region,
+		Endpoints: installConfig.Config.Platform.AWS.ServiceEndpoints,
+	})
```

Comment:
You mentioned that you were still open to suggestions despite this having merged: I would suggest creating a function on `metadata` that can create or get a client without passing `EndpointOptions`. Similar to the current `getSession`:

https://github.com/openshift/installer/blob/3b1baa02860f3e0e190ffe9b076c133a5ce5b581/pkg/asset/machines/aws/instance_types.go#L22

we would have something that is just:

`ec2Client, err := meta.EC2Client(ctx)`

This will be beneficial whenever we need to construct new clients in the asset package (which is the most common use case), we know it will be correct, and users don't need to worry about correctly configuring endpoints or fips